web-common/src/Halogen/Monaco.purs

```diff
@@ -228,12 +228,15 @@ handleQuery (GetModelMarkers f) = do
       pure $ f markers
 
 handleQuery (GetDecorationRange decoratorId f) = do
-  withEditor \editor -> do
-    liftEffect do
+  mEditor <- H.gets _.editor
+  case mEditor of
+    Nothing -> pure Nothing
+    Just editor -> do
       model <- liftEffect $ Monaco.getModel editor
-      let
-        decoRange = Monaco.getDecorationRange model decoratorId
-      pure $ f decoRange
+      mRange <- liftEffect $ Monaco.getDecorationRange model decoratorId
+      case mRange of
+        Nothing -> pure Nothing
+        Just decoRange -> pure $ Just $ f decoRange
```

Comment:
It kind of calls for a `MaybeT` monad, but I don't understand them that well, last time it took me a while to get it right with the help of @shmish111. I am happy to leave it like this.

Reply:
`Maybe` monads are very simple, but the fact you have to deal with other monads as well with the `liftEffect` makes things more tricky

Reply:
it's screaming out for a `MaybeT`!